modules/sync-narration.md

```diff
@@ -0,0 +1,200 @@
+
+While the main use of this Module is an audio narration synchonized with text highlights in a textual publication, it may also be used "in reverse mode" for enhancing an audiobook experience with textual subtitles.
+
+## Status of this document
+
+This is a draft document from the Readium community. It is currently implemented in Readium Desktop.
+
+An useful evolution would be to extend this structure with a proper support for still and moving images. A textual publication could then be enhanced with the apparition of small illustrations or sign-language videos explaining words or utterances, upon manual selection of those textual segments. This feature would be very useful for cognitive disabled people, but also for children or people learning a new language. Such feature also has the advantage of being 100% descriptive (no javascript involved in the publication).
+
+
+## Structure of a Synchronized Narration document
+
+The Synchronized Narration object is defined as:
+
+| Key  | Definition | Format | Required? |
+| ---- | -----------| -------| ----------|
+| `textRef`   |  The absolute of relative URL of the textual resource which participates to the synchronized narration | URL | At the top level only |
+| `audioRef`  |  The absolute of relative URL of the audio resource which participates to the synchronized narration | URL | At the top level only |
+| `narration` | A recursive array of synchronization objects | Array of Synchronization Item or Sub Narration | Yes |
+
+
+The Sub Narration object is defined as:
+
+| Key  | Definition | Format | Required? |
+| ---- | -----------| -------| ----------|
+| `role`      | Semantic information relative to the role of the object in the publication | [EPUB Structural Semantics](#references) | No |
+| `narration` | A recursive array of synchronization objects | Array of Synchronization Item or Sub Narration | Yes |
+
+
+The Synchronization Item object is defined as:
+
+| Key  | Definition | Format | Required? |
+| ---- | -----------| -------| ----------|
+| `text`  | The fragment identifier of the textual fragment which participates to the synchronized narration | URI Fragment Identifier | Yes |
+| `audio` | The fragment identifier of the audio fragment which participates to the synchronized narration | Media Fragment URI | Yes |
+
+**Notes:**
+
+Structural semantics offer a way to selectively filter out content based on semantics ("skippability" in the DAISY world)
+
+The recursive structure offers a way to jump out of complex structures and back into the reading flow ("escapability" in the DAISY world).
+
+In case we decide to extend the structure to image and video, using `image` and `video` would be consistent with the [latest work of the W3C CG](https://w3c.github.io/sync-media-pub/sync-media.html). 
+
+## Example
+
+```json
+{
+  "textRef": "/text/chapter1.html",
+  "audioRef": "/audio/chapter1.mp3",
```

Comment:
The term `ref` is not used anywhere else in RWPM, I think. How about `href`?

Reply:
Also: the leading slash should be removed.

Reply:
Personally I feel uneasy about hoisting "base" URLs into a higher level of the data serialisation hierarchy, for different media types (e.g. text and audio ... but what about others, and what about cases where not all resources of a given media type have the same base URL?). Also, this competes with the implicit base URL of the JSON resource itself, and its 'self' override (if any) ... and potentially its 'base' as well (see prior discussion liked below)

https://github.com/readium/architecture/discussions/109#discussioncomment-204262

Reply:
> Also, this competes with the implicit base URL of the JSON resource itself, and its 'self' override (if any) ... and potentially its 'base'

When I mentioned a "base URL" during last call I was actually referring to a `self` link, so that relative hrefs would work like in a regular RWPM.

> but what about others, and what about cases where not all resources of a given media type have the same base URL?)

If on different servers, then I guess they need to be absolute URLs.